File: readability/readability.c
//declare librarys
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

//start of main
int main(void)
{
//Prompt user for string
    string text = get_string("Text: ");

//Count number of letters
    float letters = 0;
    float words = 1;
    float sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //Count number of letters by checking for their ascii value equivalent
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters = letters + 1;
        }

        //Count Number of words by counting spaces
        if (text[i] == 32)
        {
            words = words + 1;
        }

        //Count number of sentences by counting '.', '!', '?'
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences = sentences + 1;
        }
    }


//Use Coleman-Liau formula rounded to nearest integer to determine grade level.
//Determine average number of letters per 100 words, L.
    float L = letters / words * 100;
//Determine the average number of sentences per 100 words, S.
    float S = sentences / words * 100;

//The final determination is as follows:
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }

    return 0;
}

File: sentimental-readability/readability.py
# TODO

def main():
    user_string = input("Text:")

    words = count_words(user_string)
    #print("words:", words)
    sentences = count_sentences(user_string)
    #print("sentences: ", sentences)
    letters = letter_count(user_string)
    #print("letters: ", letters)
    score = Coleman_Liau(letters, words, sentences)

    grade_level(round(score))


##Count the number of spaces in string
def count_words(string):
    spaces = 1
    for i in range(0, len(string)):
        if (string[i] == " "):
            spaces += 1
    #print(spaces)
    return spaces


##Count the number of sentences per string
def count_sentences(string):
    punctuation = 0
    for i in range (0, len(string)):
        if (string[i] == '.' or string[i] =='?' or string[i] == '!'):
            punctuation +=1
    #print(punctuation)
    return punctuation


def letter_count(string):
    alpha_count =0
    for i in range(0, len(string)):
        if((ord(string[i]) >= 65 and ord(string[i])<=90) or (ord(string[i]) >= 97 and ord(string[i]) <= 122)):
            alpha_count += 1
    return alpha_count

"""##Count additional punctuation
def other_punct(string):
    punctuation = 0
    for i in range(0, len(string)):
        if (string[i] == "," or string[i] == ';' or string[i] ==':' or string[i] == '"'):
            punctuation+=1
    return punctuation"""


def Coleman_Liau(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)

def grade_level(index):

    if (index > 16):
        print("Grade 16+")
        return
    elif (index < 1):
        print("Before Grade 1")
        return
    else:
        print(f"Grade {index}")
        return

if __name__ == "__main__":
    main()
